demo/demo.js

```diff
@@ -89,72 +75,42 @@ class App extends PureComponent {
+    return new Promise((resolve, reject) => {
       canvas.toBlob(blob => {
-        blob.name = fileName; // eslint-disable-line no-param-reassign
-        window.URL.revokeObjectURL(this.fileUrl);
-        this.fileUrl = window.URL.createObjectURL(blob);
-        resolve(this.fileUrl);
+        if (!blob) {
+          //reject(new Error('Canvas is empty'));
+          console.error('Canvas is empty');
+          return;
+        }
+        blob.name = fileName;
+        window.URL.revokeObjectURL(fileUrl);
+        const objUrl = window.URL.createObjectURL(blob);
+        setFileUrl(objUrl);
+        resolve(objUrl);
       }, 'image/jpeg');
     });
   }
 
-  makeClientCrop(crop) {
-    if (this.imageRef && crop.width && crop.height) {
-      this.getCroppedImg(this.imageRef, crop, 'newFile.jpeg').then(croppedImageUrl =>
-        this.setState({ croppedImageUrl })
-      );
-    }
-  }
-
-  renderVideo = () => (
-    <video
-      autoPlay
-      loop
-      style={{ display: 'block', maxWidth: '100%' }}
-      onLoadStart={e => {
-        // You must inform ReactCrop when your media has loaded.
-        e.target.dispatchEvent(new Event('medialoaded', { bubbles: true }));
-      }}
-    >
-      <source src={mp4Url} type="video/mp4" />
-    </video>
-  );
-
-  renderSelectionAddon = () => <input placeholder="Type something" />;
-
-  render() {
-    const { croppedImageUrl } = this.state;
-
-    return (
-      <div className="App">
-        <div>
-          <input type="file" onChange={this.onSelectFile} />
-        </div>
-        {this.state.src && (
-          <>
-            <ReactCrop
-              // renderComponent={this.renderVideo()}
-              src={this.state.src}
-              crop={this.state.crop}
-              ruleOfThirds
-              // circularCrop
-              onImageLoaded={this.onImageLoaded}
-              onComplete={this.onCropComplete}
-              onChange={this.onCropChange}
-              onDragStart={this.onDragStart}
-              onDragEnd={this.onDragEnd}
-              // renderSelectionAddon={this.renderSelectionAddon}
-              // minWidth={100}
-              minHeight={100}
-            />
-            <button onClick={this.onChangeToIncompleteCropClick}>Change to incomplete aspect crop</button>
-          </>
-        )}
-        {croppedImageUrl && <img alt="Crop" src={croppedImageUrl} />}
+  return (
+    <div className="App">
+      <div>
+        <input type="file" accept="image/*" onChange={onSelectFile} />
       </div>
-    );
-  }
+      {src && (
+        <ReactCrop
+          src={src}
+          crop={crop}
+          ruleOfThirds
+          onImageLoaded={onImageLoaded}
+          onComplete={onCropComplete}
+          onChange={onCropChange}
+        />
+      )}
+      {croppedImageUrl && (
+        <img alt="Crop" style={{ maxWidth: '100%' }} src={croppedImageUrl} />
+      )}
+    </div>
+  );
 }
 
 ReactDOM.render(<App />, cropEditor);
```

Comment:
CropEditor does not exist.
I used:
ReactDOM.render(<App />, document.getElementById("root"));